gpu4pyscf/gto/int3c1e_ip.py

```diff
@@ -264,7 +260,7 @@ def get_int3c1e_ip2_density_contracted(mol, grids, charge_exponents, dm, intopt)
 
             charge_exponents_pointer = c_null_ptr()
             if charge_exponents is not None:
-                charge_exponents_pointer = charge_exponents[i_grid_split : i_grid_split + ngrids_of_split].data.ptr
+                charge_exponents_pointer = charge_exponents[p0:p1].data.ptr
```

Comment:
create an array for `charge_exponents[p0:p1]`. This pointer refers to the temporary array, which is a potential memory leak

Reply:
Additionally, assert c-contiguous for `charge_exponents` and `grids`